src/table/hooks/use-scroll-listener.ts

```diff
@@ -1,9 +1,12 @@
 import { useEffect } from 'react';
 import { handleHorizontalScroll } from '../utils/handle-scroll';
 
-const useScrollListener = (tableContainerRef: React.MutableRefObject<HTMLDivElement>, direction: string) => {
+const useScrollListener = (
+  tableContainerRef: React.MutableRefObject<HTMLDivElement | undefined>,
+  direction?: string
+) => {
   useEffect(() => {
-    const memoedContainer = tableContainerRef.current;
+    const memoedContainer = tableContainerRef?.current;
```

Comment:
No need for the chaining. It is current that can be undefined

Reply:
why?
![Screenshot 2022-08-18 at 14 37 33](https://user-images.githubusercontent.com/4471489/185396426-c7e918b5-b17a-4778-8a9b-5660e16f8abb.png)

Reply:
yeah bit tableContainerRef is never undefined, it is current that is undefined

Reply:
You are not doing it for the other hook and that works right?

Reply:
I guess the same pattern will be used elsewhere when I reach more components. For example:

![Screenshot 2022-08-18 at 16 12 12](https://user-images.githubusercontent.com/4471489/185416576-3647fe70-2420-4b5d-91ce-327710e81478.png)
.

Reply:
yes, you already typed typed that one in this PR? and does it work without the optional chaining?

Reply:
Sorry, now I see. I get confused with `React.MutableRefObject<HTMLDivElement | undefined>` and `React.MutableRefObject<HTMLDivElement> | undefined`. You are right, optional changing is not needed here.

Reply:
Fixed